### PR TITLE
[hydra-typegen] and yml config support

### PR DESCRIPTION
**Pull request by dzhelezov**
Use YML file along with cli flags for typegen runs.

Example:
```yml
metadata:
  source: wss://kusama-rpc.polkadot.io
  blockHash: '0x45eb7ddd324361adadd4f8cfafadbfb7e0a26393a70a70e5bee6204fc46af62e'
events:
  - Balances.Transfer
calls:
  - Balances.transfer
outDir: ./generated
```

### CODE REVIEW

File: packages/hydra-typegen/src/util.ts [hunk 1]
```diff
@@ -10,17 +10,14 @@ export function readTemplate(template: string): string {
...
+  return fs.readFileSync(location, 'utf-8')
 }
 
-export function readJson(
-  source: string,
-  root: string = process.cwd()
-): Record<string, unknown> {
-  return JSON.parse(readFile(source, root))
+export function readJson(source: string): Record<string, unknown> {
+  return JSON.parse(path.resolve(source))
```
Comment (metmirr) [inline packages/hydra-typegen/src/util.ts:20]: I believe it should be `JSON.parse(readFile(path.resolve(source)))`

File: packages/hydra-typegen/src/commands/typegen/index.ts [hunk 2]
```diff
@@ -120,34 +149,43 @@ types don't much the metadata definiton`,
...
   }
 
-  generate(config: IConfig): void {
-    const { dest, customTypes } = config
+  async buildGeneratorConfig(config: IConfig): Promise<GeneratorConfig> {
+    const { outDir, customTypes } = config
 
-    debug(`Output dir: ${dest}`)
-    fs.mkdirSync(dest, { recursive: true })
+    // if (customTypes) {
```
Comment (metmirr) [inline packages/hydra-typegen/src/commands/typegen/index.ts:168]: I needed to uncomment this code to make it work with my yaml file.

File: packages/hydra-typegen/src/util.ts (hunk 1 above)
Comment (dzhelezov) [reply packages/hydra-typegen/src/util.ts]: Fixed

File: packages/hydra-typegen/src/commands/typegen/index.ts (hunk 2 above)
Comment (dzhelezov) [reply packages/hydra-typegen/src/commands/typegen/index.ts]: Fixed

**Pull request merged**